common/djangoapps/edxmako/template.py

```diff
@@ -13,47 +13,89 @@
 #   limitations under the License.
 
 from django.conf import settings
+from django.template import Context, engines
 from mako.template import Template as MakoTemplate
+from six import text_type
 
-import edxmako
-from edxmako.request_context import get_template_request_context
-from edxmako.shortcuts import marketing_link
+from . import Engines, LOOKUP
+from .request_context import get_template_request_context
+from .shortcuts import is_any_marketing_link_set, is_marketing_link_set, marketing_link
 
+KEY_CSRF_TOKENS = ('csrf_token', 'csrf')
 
-# TODO: We should make this a Django Template subclass that simply has the MakoTemplate inside of it? (Intead of inheriting from MakoTemplate)
 
-
-class Template(MakoTemplate):
+class Template(object):
     """
-    This bridges the gap between a Mako template and a djano template. It can
-    be rendered like it is a django template because the arguments are transformed
+    This bridges the gap between a Mako template and a Django template. It can
+    be rendered like it is a Django template because the arguments are transformed
     in a way that MakoTemplate can understand.
     """
 
     def __init__(self, *args, **kwargs):
         """Overrides base __init__ to provide django variable overrides"""
-        if not kwargs.get('no_django', False):
-            kwargs['lookup'] = edxmako.LOOKUP['main']
-        super(Template, self).__init__(*args, **kwargs)
+        self.engine = kwargs.pop('engine', engines[Engines.MAKO])
+        if len(args) and isinstance(args[0], MakoTemplate):
+            self.mako_template = args[0]
+        else:
+            kwargs['lookup'] = LOOKUP['main']
```

Comment:
Is `LOOKUP['main']` guaranteed to exist here?

Reply:
In practice, yes.  It doesn't feel like throwing an exception from `__init__` or passing `None` into the `MakoTemplate` constructor would be any better at debugging future failures around this than just bubbling up the `KeyError`; do you have any better suggestions?

Reply:
Nope - I support just letting the KeyError bubble up if it ever happens. We'd see it in the logs.